testing/util/lightclient.go

```diff
@@ -28,1049 +37,833 @@ type TestLightClient struct {
-	ctx := context.Background()
+func NewTestLightClient(t *testing.T, forkVersion int, options ...LightClientOption) *TestLightClient {
+	l := &TestLightClient{T: t, version: forkVersion}
 
-	slot := primitives.Slot(params.BeaconConfig().CapellaForkEpoch * primitives.Epoch(params.BeaconConfig().SlotsPerEpoch)).Add(1)
-	if increaseAttestedSlotBy > 0 {
-		slot = slot.Add(uint64(increaseAttestedSlotBy))
+	for _, option := range options {
+		option(l)
 	}
 
-	attestedState, err := NewBeaconStateCapella()
-	require.NoError(l.T, err)
-	err = attestedState.SetSlot(slot)
-	require.NoError(l.T, err)
-
-	finalizedBlock, err := blocks.NewSignedBeaconBlock(NewBeaconBlockCapella())
-	require.NoError(l.T, err)
-	finalizedBlock.SetSlot(primitives.Slot(params.BeaconConfig().CapellaForkEpoch * primitives.Epoch(params.BeaconConfig().SlotsPerEpoch)))
-	finalizedHeader, err := finalizedBlock.Header()
-	require.NoError(l.T, err)
-	finalizedRoot, err := finalizedHeader.Header.HashTreeRoot()
-	require.NoError(l.T, err)
-
-	require.NoError(l.T, attestedState.SetFinalizedCheckpoint(&ethpb.Checkpoint{
-		Epoch: params.BeaconConfig().CapellaForkEpoch,
-		Root:  finalizedRoot[:],
-	}))
-
-	parent := NewBeaconBlockCapella()
-	parent.Block.Slot = slot
-
-	signedParent, err := blocks.NewSignedBeaconBlock(parent)
-	require.NoError(l.T, err)
-
-	parentHeader, err := signedParent.Header()
-	require.NoError(l.T, err)
-	attestedHeader := parentHeader.Header
-
-	err = attestedState.SetLatestBlockHeader(attestedHeader)
-	require.NoError(l.T, err)
-	attestedStateRoot, err := attestedState.HashTreeRoot(ctx)
-	require.NoError(l.T, err)
-
-	// get a new signed block so the root is updated with the new state root
-	parent.Block.StateRoot = attestedStateRoot[:]
-	signedParent, err = blocks.NewSignedBeaconBlock(parent)
-	require.NoError(l.T, err)
-
-	state, err := NewBeaconStateCapella()
-	require.NoError(l.T, err)
-	err = state.SetSlot(slot)
-	require.NoError(l.T, err)
-
-	parentRoot, err := signedParent.Block().HashTreeRoot()
-	require.NoError(l.T, err)
-
-	var signedBlock interfaces.SignedBeaconBlock
-	if blinded {
-		block := NewBlindedBeaconBlockCapella()
-		block.Block.Slot = slot
-		block.Block.ParentRoot = parentRoot[:]
+	switch l.version {
+	case version.Altair:
+		return l.setupTestAltair()
+	case version.Bellatrix:
+		return l.setupTestBellatrix()
+	case version.Capella:
+		return l.setupTestCapella()
+	case version.Deneb:
+		return l.setupTestDeneb()
+	case version.Electra:
+		return l.setupTestElectra()
+	default:
+		l.T.Fatalf("unknown version %d", l.version)
+		return nil
+	}
+}
 
-		var trueBitNum uint64
-		if supermajority {
-			trueBitNum = (params.BeaconConfig().SyncCommitteeSize / 3 * 2) + 1
-		} else {
-			trueBitNum = params.BeaconConfig().MinSyncCommitteeParticipants
-		}
-		for i := uint64(0); i < trueBitNum; i++ {
-			block.Block.Body.SyncAggregate.SyncCommitteeBits.SetBitAt(i, true)
+// WithBlinded Specifies whether the signature block is blinded or not
```

Comment:
```suggestion
// WithBlinded specifies whether the attested block is blinded or not
```
Is that what you meant? I am not sure what a "Signature block" is

Reply:
No it's for the signature block. the block immediately after the attested block, which holds the signatures for the attested block.